### PR TITLE
final submission. No step 11.

### PR DESCRIPTION
**Pull request by chrispriiice**
@natehedgeman

### CODE REVIEW

**Comment by hedgemanNate** [comment]
### Step 1
`let ethanHunt = (coverName: "Ethan Hunt", realName: "Tom Cruise", accessLevel: 8, compromised: true)
let jimPhelps = (coverName: "Jim Phelps", realName: "Jon Voight", accessLevel: 9, compromised: false)
let clairePhelps = (coverName: "Claire Phelps", realName: "Emmanuelle Beart", accessLevel: 5, compromised: false)
let eugeneKittridge = (coverName: "Eugene Kittridge", realName: "Henry Czerny", accessLevel: 10, compromised: true)
let franzKrieger = (coverName: "Franz Krieger", realName: "Jean Reno", accessLevel: 4, compromised: false)
let lutherStickell = (coverName: "Luther Stickell", realName: "Ving Rhames", accessLevel: 4, compromised: false)
let sarahDavies = (coverName: "Sarah Davies", realName: "Kristin Scott Thomas", accessLevel: 5, compromised: true)
let maxRotGrab = (coverName: "Max RotGrab", realName: "Vanessa Redgrave", accessLevel: 4, compromised: false)
let hannahWilliams = (coverName: "Hannah Williams", realName: "Ingeborga Dapkūnaitė", accessLevel: 5, compromised: true)
let jackHarmon = (coverName: "Jack Harmon", realName: "Emilio Estevez", accessLevel: 6, compromised: true)
let frankBarnes = (coverName: "Frank Barnes", realName: "Dale Dye", accessLevel: 9, compromised: false)`

Not a big deal and not mandatory, but I think it would be a good idea to name the agent constants "Agent1, Agent2..." or something a little more abstract than their names. Since Swift is an object-oriented programming language, it would be good to start thinking about what you are creating as empty placeholders/vessels than actual individuals. Again, you did nothing wrong here just a suggestion. 

### Step 2
Great!

###  Step 3
Great! Something that I have to constantly work on is my spacing. Nitpicking here: line 58 is an open line that could be erased. Again serious Nitpicking but something to keep in the back of your mind. 

### Step 4
Awesome!

### Step 5
Great job on the code. Another way to handle that conditional statement in the for loop would have looked something like:

`for agent in agents {
        if !agent.compromised {
            print(agent.coverName)
            cleanAgents.append(agent)
        }
    }`

The  `if !agent.compromised` means, IF the agent IS NOT compromised. In coditional statements, the "!" means "IS NOT".

Also, it looks like when you were typing out the agent tuples, you missed typed some of the agents' details.

`let jimPhelps = (coverName: "Jim Phelps", realName: "Jon Voight", accessLevel: 9, compromised: false)`

For example, Jim Phelps is actually a compromised agent. Not a big deal here but be mindful to type information accurately when you copying. This wasn't something that would break the code here but in other cases inaccuracy could. Even though you need practiced TYPING could since it is all new. Try to look for opportunities that you can copy and paste code that is given to you. Not just for accuracy but also for time. When I did this project yesterday, I think copying pasting those agent detail probably saved me about 10 minutes alone. 

### Step 6
Awesome!!

### Step 7
Great job! You just forgot to add "level:" to your string. So it just needed whats in between the two arrowheads (^, level:  ^) 

`func findHighRisk () {
    for agent in agentArray {
        if agent.accessLevel >= 8 && agent.compromised == true {
            print("\(agent.realName)^, level:  ^\(agent.accessLevel) **WARNING** **COMPROMISED**")
        } else if agent.accessLevel >= 8 {
            print("\(agent.realName) \(agent.accessLevel)")
        }
    }
}`

### Step 8
Perfect

### Step 9
It works and THAT is the most important part. But a simplified solution could look like this.

`func determineLevels() {
    var lowLevel = 0
    var midLevel = 0
    var highLevel = 0
    
    for agent in agents {
        switch agent.accessLevel {
        case 1...4:
            lowLevel += 1
        case 5...7:
            midLevel += 1
        default:
            highLevel += 1
        }
    }
    
    print("\(lowLevel) low level agents, \(midLevel) mid level agents, and \(highLevel) high level agents")
}`

Basically instead of using arrays to hold the number of agents that fit an access level range, you just create variables to hold the count itself. And instead of appending the agent itself to an array, you simply add 1 to the count when you find a matching agent. So when you code thinks about how to accomplish the goal as simple as possible. It will save time, keystrokes and make it easier for others to understand.

But like I said your way WORKS as well. And that is what is most important.  

### Step 10
Perfect!
